SAMPLE_FORMAT.md

```diff
@@ -25,19 +57,128 @@ specifically mentioned in the format, samples should make best-effort attempts i
 * __Idiomatic__ - examples should make best attempts to remain idiomatic and encourage good 
   practices that are specific to a language or practice. 
 
-## Format Guidelines
+## Build Tools
+
+All new samples should build and run integration tests with [Maven](https://maven.apache.org/).
+[Gradle](https://gradle.org/) support is optional as we don't yet have regular testing.
+
+## Testing
+Snippets should have integration tests that should verify the snippet works and compiles correctly.
+Creating mocks for these tests are optional. These tests should capture output created by the
+snippet to verify that it works correctly. See the tests in the canonical for an example of how to
+do this correctly. 
+
+* Test Library: [JUnit4](https://junit.org/junit4/)
+* Test Runner: [Maven Failsafe plugin](https://maven.apache.org/surefire/maven-failsafe-plugin/)
+and [Maven Surefire plugin](https://maven.apache.org/surefire/maven-surefire-plugin/).
 
+As an example, the following test code shows how we test a region tag called `region_tag`:
```

Comment:
Is it worth explaining what region tags are here? (Perhaps for external users and/or Googlers outside our team.)

Reply:
I'll put something in, but generally no.